modules/monitor-data-collection-rule/main.tf

```diff
@@ -0,0 +1,247 @@
+      for_each = var.destinations["monitor_account"] != null ? var.destinations["monitor_account"] : []
+      content {
+        name               = monitor_account.value["name"]
+        monitor_account_id = monitor_account.value["monitor_account_id"]
+      }
+    }
+    dynamic "storage_blob" {
+      for_each = var.destinations["storage_blob"] != null ? var.destinations["storage_blob"] : []
+      content {
+        name               = storage_blob.value["name"]
+        storage_account_id = storage_blob.value["storage_account_id"]
+        container_name     = storage_blob.value["container_name"]
+      }
+    }
+    dynamic "storage_blob_direct" {
+      for_each = var.destinations["storage_blob_direct"] != null ? var.destinations["storage_blob_direct"] : []
+      content {
+        name               = storage_blob_direct.value["name"]
+        storage_account_id = storage_blob_direct.value["storage_account_id"]
+        container_name     = storage_blob_direct.value["container_name"]
+      }
+    }
+    dynamic "storage_table_direct" {
+      for_each = var.destinations["storage_table_direct"] != null ? var.destinations["storage_table_direct"] : []
+      content {
+        name               = storage_table_direct.value["name"]
+        table_name         = storage_table_direct.value["table_name"]
+        storage_account_id = storage_table_direct.value["storage_account_id"]
+      }
+    }
+  }
+
+  dynamic "stream_declaration" {
+    for_each = var.stream_declaration != null ? [var.stream_declaration] : []
+    content {
+      stream_name = stream_declaration.value["stream_name"]
+      dynamic "column" {
+        for_each = stream_declaration.value["column"]
+        content {
+          name = column.value["name"]
+          type = column.value["type"]
+        }
+      }
+    }
+  }
+
+  dynamic "identity" {
+    for_each = var.identity != null ? [var.identity] : []
+    content {
+      type         = identity.value["type"]
+      identity_ids = identity.value["identity_ids"]
+    }
+  }
+}
+
+# azurerm_monitor_data_collection_rule_association doesn't support resource of type
+# 'Microsoft.OperationalInsights/workspaces'
+# https://github.com/hashicorp/terraform-provider-azurerm/issues/20637
+
+#resource "azurerm_monitor_data_collection_rule_association" "dcr_association" {
+#  name                    = "${var.name}-association"
+#  description             = var.description
+#  target_resource_id      = var.target_resource_id
+#  data_collection_rule_id = azurerm_monitor_data_collection_rule.dcr.id
+#}
+
+# TODO: add validations for resource_group_name and target_resource_id to prevent potential CLI injection.
+resource "null_resource" "dcr_association" {
+  provisioner "local-exec" {
+    command = "az monitor log-analytics workspace update --resource-group ${var.resource_group_name} --workspace-name ${var.target_resource_id} --data-collection-rule \"${azurerm_monitor_data_collection_rule.dcr.id}\""
```

Comment:
Is this to "deploy" the DCR?

I would think applying "azurerm_monitor_data_collection_rule" would deploy the rule to the ALA workspace.

Reply:
It doesn't 😢

Reply:
Check this out https://github.com/hashicorp/terraform-provider-azurerm/issues/20637

Reply:
Can you add some comment for line 242? I'm certain that in 6 months, I'm going to forget and ask myself, why are we doing this? Thanks bro.

Reply:
Done 👍🏻